create-publication-from-scopus/src/main/java/no/sikt/nva/scopus/ScopusConverter.java

```diff
@@ -0,0 +1,165 @@
+package no.sikt.nva.scopus;
+
+import static no.sikt.nva.scopus.ScopusConstants.DOI_OPEN_URL_FORMAT;
+import jakarta.xml.bind.JAXB;
+import java.io.StringWriter;
+import java.net.URI;
+import java.util.Collection;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Collectors;
+import no.scopus.generated.AuthorGroupTp;
+import no.scopus.generated.AuthorTp;
+import no.scopus.generated.CollaborationTp;
+import no.scopus.generated.DocTp;
+import no.scopus.generated.ItemidTp;
+import no.scopus.generated.TitletextTp;
+import no.scopus.generated.YesnoAtt;
+import no.unit.nva.metadata.CreatePublicationRequest;
+import no.unit.nva.model.AdditionalIdentifier;
+import no.unit.nva.model.Contributor;
+import no.unit.nva.model.EntityDescription;
+import no.unit.nva.model.Identity;
+import no.unit.nva.model.Reference;
+import nva.commons.core.JacocoGenerated;
+import nva.commons.core.paths.UriWrapper;
+
+public class ScopusConverter {
+
+    @JacocoGenerated
+    public ScopusConverter() {
```

Comment:
This class is a typical utility class; we aren't instantiating objects, we're simply utilising the methods. Make the entry method static, follow the rabbit down the hole (fortunately your IDE solves this for you).

At the same time, utility classes tend to indicate that something isn't entirely right in object-oriented code (google this and tell me if you think it is pedantic). You might therefore consider renaming the class ScopusPublicationRequest, implementing a constructor that takes a single DocTp as argument, and implementing a public toCreatePublicationRequest method.

That said, I also feel that CreatePublicationRequest is actually an interface and that we are implementing it here.

Reply:
My suggestion is basically the same as Rurik's suggestion :P

Reply:
Fixed to protected class with DocTp in constructor.

I can see that the original implementation is not great object orientation. But having code that conforms to PURE principles is also valuable.

Reply:
By PURE principles, do you mean referential transparency?

Reply:
https://en.wikipedia.org/wiki/Pure_function